### ISSUE TITLE
got 「unknown」 type when using Instance<typeof SomeModel>

### ISSUE THREAD

**Issue by JLUiceman** [mobxjs/mobx-state-tree#2217]
**_Bug report_**

-   [x] I've checked documentation and searched for existing issues [and discussions](https://github.com/mobxjs/mobx-state-tree/discussions)
-   [x] I've made sure my project is based on the latest MST version
-   [x] Fork [this](https://codesandbox.io/s/y64pzxj01) code sandbox or another minimal reproduction.

**Sandbox link or minimal reproduction code**
I try to get the type of model by using Instance method just like the doc recommended, but the result confused me:

when i define a model like the pic below, it works fine
<img width="666" alt="截屏2024-10-22 15 27 30" src="https://github.com/user-attachments/assets/ecdebc1e-a72b-4302-8a4f-3a845b4f79ca">

but when i define a model in this way, i got a unknown type
<img width="687" alt="截屏2024-10-22 15 22 14" src="https://github.com/user-attachments/assets/a0a96945-1e57-4982-a57f-7827d1a93cf2">

Is there anyway i can avoid this or i just missed some key information when reading the doc?

**Comment by kuguarpwnz** [comment]
type resolution does not work for me in WebStorm
<img width="315" alt="image" src="https://github.com/user-attachments/assets/5b0baa66-2bbb-4e8d-96d1-50b6bb648b24">

although it works in vs code
<img width="548" alt="image" src="https://github.com/user-attachments/assets/3c8dd266-49bb-41f4-a202-17116df4c14c">

**Comment by JLUiceman** [comment]
> @JLUiceman are you on the latest version? I wasn't able to reproduce :(
> 
> ![CleanShot 2024-10-22 at 09 44 39@2x](https://private-user-images.githubusercontent.com/833576/378865802-aa5351ea-fe2f-40f0-9686-2ebaa0981c47.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk2NDc4NTAsIm5iZiI6MTcyOTY0NzU1MCwicGF0aCI6Ii84MzM1NzYvMzc4ODY1ODAyLWFhNTM1MWVhLWZlMmYtNDBmMC05Njg2LTJlYmFhMDk4MWM0Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyM1QwMTM5MTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNWMwMWVlNzQwMTkyOGQxN2Q5NmZlMzQ3ZGM3NGJjMGUwZjVlYWI0NzdiOTdjMmJjMTQ0ZjcyNzA4NDczMmQ3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.MaKV3v5rFFdm9oI3--RHbSyal4z8C_d24aeQoXoo2cg)

@thegedge Thanks for reply! Yes, project's current mst version is 6.0.1;
@kuguarpwnz kuguarpwnz's comment inspired me; I tried to run the same code in VSCode, and it works fine :). I'm curious why there is such a difference.

**Comment by thegedge** [comment]
Great question!

If I had to guess, it's one of three things:

1. A language server other than `tsserver` that's behaving differently.
2. The language server not properly loading up tsconfig.
3. Wrong version of typescript is evaluating things inside the language server.

**Comment by JLUiceman** [comment]
> That being said, I'll flag this as a perhaps a small docs update where we could have a callout for issues with types when using WebStorm. If you do figure out what the issue is, though, drop a comment here :)

Yep, I will also continue to follow up on this issue and try to find out the possible reasons. If there are further conclusions, I will comment here.
Thanks for your patient explanation, and hope we can got a solution:)

**Comment by jaryway** [comment]
![image](https://github.com/user-attachments/assets/9768e63b-5ec2-4b4f-8a8b-8cc91d05f28e)
![image](https://github.com/user-attachments/assets/d78a48bb-77af-4ad5-8ac8-b1055002c342)
![image](https://github.com/user-attachments/assets/b397c8bf-ada1-4125-9225-c5c549a14f45)
